File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        holder, seeker = 0, 0
        n = len(nums)
        while seeker < n:
            if nums[seeker] != 0:
                nums[seeker], nums[holder] = nums[holder], nums[seeker]
                holder += 1
            seeker += 1
        


        

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int left = -1, right = -1;
        for (int i = 0; i < nums.length; i++){
            if (nums[i] == 0){
                left = i;
                break;
            }
        }
        if (left == -1) return;
        right = left + 1;
        while (right < nums.length){
            if (nums[right] != 0){
                int temp = nums[right];
                nums[right] = nums[left];
                nums[left] = temp;
                left ++;
            } else {
                right ++;
            }
        }
        
    }
}